narwhal/types/src/primary.rs

```diff
@@ -1183,6 +1225,278 @@ impl CertificateV1 {
     }
 }
 
+// Holds AggregateSignatureBytes but with the added layer to specify if the
+// signature was verified via a leader, verified directly, unverified or
+// unsigned. This will be used to take advantage of the certificate chain that
+// is formed via the DAG by only verifying the leaders of the certificate chain
+// when they are fetched from validators during catchup.
+#[derive(Clone, Serialize, Deserialize, MallocSizeOf, Debug)]
+pub enum AggregateSignatureState {
+    VerifiedViaLeader(AggregateSignatureBytes),
+    VerifiedDirectly(AggregateSignatureBytes),
+    Unverified(AggregateSignatureBytes),
+    Unsigned(AggregateSignatureBytes),
```

Comment:
I would suggest to add some comments to explain what every state means here

Reply:
also I see the `AggregateSignatureState` actually representing the certificate's verification state, rather than the aggregate signature it self. So maybe (just a proposal) something like `VerificationState` would be more appropriate?